mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/query/parsing/nativequery/impl/NativeQueryParser.java

```diff
@@ -339,19 +382,22 @@ public Rule JsonComposite() {
 	}
 
 	public Rule JsonObject() {
-		return Sequence(
+		return FirstOf(
+			Sequence(
 				ZeroOrMore( WhiteSpace() ).skipNode(),
 				"{ ",
 				FirstOf(
 						Sequence( Pair(), ZeroOrMore( Sequence( ", ", Pair() ) ) ),
 						Optional( Pair() )
 				).suppressNode(),
 				"} "
+			),
+			Sequence( ZeroOrMore( ANY ), builder.setInvalidJsonInPrefix( match() ), ACTION( false )  )
```

Comment:
OK, somehow I missed this change when doing my first review. It's a good idea but it needs some more love.

So I think the plan would be:
 * keep JsonObject as is. It's better this way as it has a clear meaning of what it is.
 * add a rule named `JsonObjectParameter` that contains your new stuff and wraps the `JsonObject` rule and use it wherever it should be used
 * rename `invalidJsonInPrefix` to `invalidJsonParameter` (the property and the method)

Reply:
`JsonObject` is used in `JsonComposite` and some operations rules use `JsonComposite`. Maybe it is better to wrap both this rules something like this?
```java
public Rule Insert() {
    return Sequence(
            "insert ",
            builder.setOperation( Operation.INSERT ),
            "( ",
            JsonParameter( JsonComposite() ), builder.setUpdateOrInsert( match() ),
            Optional( Sequence( ", ", JsonParameter( JsonObject() ), builder.setOptions( match() ) ) ),
            builder.setParametersValid( true ),
            ") "
    );
}

...
public Rule JsonParameter(Rule Parameter) {
    return FirstOf(
            Parameter,
            Sequence( ZeroOrMore( ANY ), builder.setInvalidJsonParameter ( match() ), ACTION( false )  )
    );
}
```

Reply:
Yes, sounds like a good idea!